File: Python/Problem3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        substring = ""
        longest = 0
        for i in range (0, len(s)):
            if s[i] in substring:
                if len(substring) > longest:
                    longest = len(substring)
                    substring = s[i]
            else:
                substring += s[i]
        return longest
        

File: TypeScript/Problem3.ts
function lengthOfLongestSubstring(s: string): number {
    let substring = "";
    let longest = 0;

    for(let i = 0; i<= s.length; i++)
    {
        if(substring.includes(s[i]))
        {
            if(substring.length > longest)
            {
                longest = substring.length;
                substring = s[i];
            }
        }
        else
        {
            substring+=s[i];
        }
    }
    return longest;
};